src/components/List/BaseList.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from "react";
+import { List, Button, Spin, Row } from "antd";
+import { useTranslation } from "react-i18next";
+
+const BaseList = ({
+    apiCall,
+    renderItem,
+    filter = {}, // replaced 'query' parameter
+    emptyFallback = <></>,
+    style = {},
+    footer = <></>
+}) => {
+    const { t } = useTranslation();
+
+    const [ loading, setLoading ] = useState(false)
+    const [ initLoading, setInitLoading ] = useState(true)
+    const [ totalPages, setTotalPages ] = useState(0)
+    const [ totalItems, setTotalItems ] = useState(0)
+    const [ items, setItems ] = useState([])
+    const [ query, setQuery ] = useState({
+        page: 1,
+        pageSize: 10,
+        fetchOnly: true,
+        ...filter
+    })
+
+    useEffect(() => {
+        apiCall(query).then(newItems => {
+            setInitLoading(false)
+            setLoading(false)
+            setTotalPages(newItems.totalPages)
+            setTotalItems(newItems.total)
+            setItems(items => [...items, ...newItems.data])
+        });
+    }, [query, apiCall]);
+
+    const loadMoreData = () => {
```

Comment:
This seems to be broken, it doesn't make the request to page to the next items of the list